### PR TITLE
Rails Api backend

### PR DESCRIPTION
**Pull request by mohashyne**
### This pull request establishes the foundation of the 'hello-rails-back-end' project by setting up a new Rails API application, configuring Git version control, integrating a PostgreSQL database, and implementing a messages table with a set of five unique greetings. Additionally, it introduces an API endpoint to retrieve a random greeting from the database.
Changes Made

- [ ]     Created a new Rails API app named 'hello-rails-back-end'.
- [ ]     Initialized the project with Git for version control.
- [ ]     Configured and set up a PostgreSQL database for data storage.
- [ ]     Implemented a 'messages' table within the database to store greetings.
- [ ]     Seeded the 'messages' table with five distinct greetings.
- [ ]     Created a new API endpoint to fetch a random greeting.

### Technical Details
- [ ]     Added necessary gems for PostgreSQL integration.
- [ ]     Created a 'messages' table migration.
- [ ]     Generated a controller to handle greeting retrieval.
- [ ]     Defined a route for the random greeting endpoint.

### Testing
- [ ]     Ensured that the API endpoint for random greetings returns a response with the expected structure and content.
- [ ]     Verified that the greetings are being retrieved randomly from the database.
- [ ]     You can use tools like Postman to test the API endpoint that fetches a random greeting.

### Next Steps
    Proceed with frontend integration to fetch and display greetings on the client side.

This pull request lays the groundwork for the 'hello-rails-back-end' project, providing essential backend functionality and data retrieval capabilities. Further development will involve connecting this backend with the front end for a complete user experience.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
+source 'https://rubygems.org'
```
Comment (gemmen29) [inline Gemfile:1]: - [ ] Kindly fix this error

image

I get this error when I run the frontend project in the console and I do not get any greeting message, but instead, I get an error message `Failed to fetch`.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @mohashyne, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile (hunk 1 above)
Comment (mohashyne) [reply Gemfile:1]: fix

**Review by topeogunleye — APPROVED** [review]
### Hi @mohashyne, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Fixed error and application is running as expected. ✔️ 
 
## [Optional] suggestions:
* [ ]  Please check my inline comments.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**